.github/actions/install-python-and-package/action.yml

```diff
@@ -44,7 +44,6 @@ runs:
         python3 -m pip install tensorflow-cpu
 
     - name: Install DIANNA
-      if: steps.cache-python-env.outputs.cache-hit != 'true'
```

Comment:
When fixing the CI for notebooks, the build for python 3.11 on macos fails unexpectedly. Error message indicates that `pytest` is not found. I figured out that it was due to the installation, in which pytest was installed for python 3 but the ci calls pytest from python via system path. Therefore, it can be fixed by simply change `run: pytest -v` to `run: python3 -m pytest -v`.

However, this build will fail again if the caching of package installation is used (not sure why...). By removing the cache, it works nicely. But this makes the CD/CI slower than before. If anyone has better solution, please let me know. Thanks.

Reply:
No other solution from me.